go-controller/pkg/ovn/controller/egressfirewall/egressfirewall.go

```diff
@@ -374,44 +398,75 @@ func (oc *EFController) sync(key string) (updateErr error) {
 	if err != nil {
 		return fmt.Errorf("invalid resource key for egress firewall: %s", key)
 	}
-	shouldRemove := false
-	shouldAddUpdate := false
+	oc.cache.LockKey(namespace)
+	defer oc.cache.UnlockKey(namespace)
+
+	existingEntry, _ := oc.cache.Load(namespace)
+	var newEntry *cacheEntry
 
 	ef, err := oc.efLister.EgressFirewalls(namespace).Get(efName)
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			return err
 		}
-		shouldRemove = true
 	} else {
-		shouldAddUpdate = true
-	}
-
-	oc.cache.LockKey(namespace)
-	defer oc.cache.UnlockKey(namespace)
+		defer func() {
+			if statusErr := oc.setEgressFirewallStatus(ef, updateErr); statusErr != nil {
+				updateErr = utilerrors.Join(updateErr, fmt.Errorf("failed to update egress firewall status %s, error: %w",
+					GetEgressFirewallNamespacedName(ef), statusErr))
+			}
+		}()
 
-	if entry, ok := oc.cache.Load(namespace); ok {
-		if entry.stale {
-			shouldRemove = true
+		activeNetwork, netErr := oc.networkManager.GetActiveNetworkForNamespace(namespace)
+		if netErr != nil {
+			if util.IsInvalidPrimaryNetworkError(netErr) {
+				// Namespace requires P-UDN, but it does not exist. Remove EF config and surface error in status.
+				updateErr = netErr
+			} else {
+				return fmt.Errorf("failed to get active network for egress firewall %s/%s namespace: %w",
+					namespace, efName, netErr)
+			}
+		} else {
+			aclLoggingLevels, logErr := oc.getNamespaceACLLogging(namespace)
+			if logErr != nil {
+				return fmt.Errorf("failed to get acl logging levels for egress firewall %s/%s: %w",
+					namespace, efName, logErr)
+			}
+			ownerController := activeNetwork.GetNetworkName() + "-network-controller"
+			newEntry = &cacheEntry{
+				pgName:            libovsdbutil.GetPortGroupName(getNamespacePortGroupDbIDs(namespace, ownerController)),
+				subnetsKey:        subnetsKeyForNetInfo(activeNetwork),
+				efResourceVersion: ef.ResourceVersion,
+				hasNodeSelector:   efHasNodeSelector(ef),
```

Comment:
you don't need to set it here though, because we can ignore hasNodeSelector in comparison: hasNodeSelector can't change without efResourceVersion also changing.
So you only need to know if existingEntry has it, and then set it for newEntry in reconcile?